app/helpers/templateGenerator.js

```diff
@@ -1,119 +1,148 @@
-      ? (`${opts.stringStart}${key}: ${JSON.stringify(modifiedValue)}`)
-      : (`${opts.stringStart0}${key}: ${JSON.stringify(modifiedValue)}`);
-    acc[0] += keyValueString;
-    return acc;
-  }, opts.accArray).join('');
+
+  const opts = options[isCode ? 'display' : 'code'];
+
+  let html = isCode ? '' : '<div class="bcc-c-code-nested">';
+
+  html += showKey ? `${isCode ? '' : '<label class="bcc-c-code-key-label bcc-u-code-secondary-color">'}"${key}": ${isCode ? '' : '</label>'}` : '';
+
+  if (typeof newValue === 'string') {
+    html += `${isCode ? '' : '<span class="bcc-c-code-editable-content bcc-u-code-primary-color">'}"${isCode ? '' : '<span contenteditable="true">'}`;
+    html += `${newValue.replace(opts.regex1, opts.string1).replace(opts.regex2, opts.string2)}`;
+    html += `${isCode ? '' : '</span>'}"${isCode ? '' : '</span>'}${isLast ? ' ' : ','}${isCode ? '' : '<br>'}`;
+  } else if (Array.isArray(newValue)) {
+    html += '[';
+    html += isCode ? '' : '<br><div class="bcc-c-code-json-array">';
+    html += newValue.map((val, index) => generateEditableJSON({
+      key: index,
+      value: val,
+      isLast: index + 1 === Object.keys(newValue).length,
+      showKey: false,
+      isCode,
+    })).join('');
+    html += isCode ? '' : '</div>';
+    html += ']';
+    html += isLast ? ' ' : ',';
+    html += isCode ? '' : '<br>';
+  } else {
+    html += '{';
+    html += isCode ? '' : '<br><div class="bcc-c-code-json-object">';
+    html += Object.entries(newValue).map(([k, v], index) => generateEditableJSON({
+      key: k,
+      value: v,
+      isLast: index + 1 === Object.keys(newValue).length,
+      isCode,
+    })).join('');
+    html += isCode ? '' : '</div>';
+    html += '}';
+    html += isLast ? ' ' : ',';
+    html += isCode ? '' : '<br>';
+  }
+
+
+  html += isCode ? '' : '</div>';
+
+  return html;
 };
 
-const getSettings = (component) => {
-  const settingsString = fs.readFileSync(`app/components/${component}/settings.json`, 'utf-8');
+const generateEditableCode = (params, isCode = false) => {
+  const keys = Object.keys(params);
+  return keys.map((key, index) => generateEditableJSON({
+    key,
+    value: params[key],
+    isLast: index + 1 === Object.keys(params).length,
+    isCode,
+  })).join('');
+};
+
+const getSettings = (name, type) => {
+  const settingsString = fs.readFileSync(`app/${type}s/${name}/settings.json`, 'utf-8');
   return JSON.parse(settingsString);
 };
 
+const writeTemplate = (template, name, type) => {
+  const directory = 'app/templates';
+  const subDirectory = `${directory}/${type}s`;
+  if (!fs.existsSync(directory)) fs.mkdirSync(directory);
+  if (!fs.existsSync(subDirectory)) fs.mkdirSync(subDirectory);
+  fs.writeFileSync(`${subDirectory}/${name}-template.njk`, template);
+};
+
 const generateTemplate = ({
-  component,
+  name,
   formParams = {},
+  templateType: type,
 }) => {
-  const settings = getSettings(component);
+  const settings = getSettings(name, type);
```

Comment:
can we destructure this in one line?
`const { componentName, params } = getSettings(name, type); `